datafusion/functions-aggregate/src/bool_and_or.rs

```diff
@@ -0,0 +1,408 @@
+}
+
+impl Accumulator for BoolOrAccumulator {
+    fn update_batch(&mut self, values: &[ArrayRef]) -> Result<()> {
+        let values = &values[0];
+        self.acc = match (self.acc, bool_or_batch(values)?) {
+            (None, ScalarValue::Boolean(v)) => v,
+            (Some(v), ScalarValue::Boolean(None)) => Some(v),
+            (Some(a), ScalarValue::Boolean(Some(b))) => Some(a || b),
+            _ => unreachable!(),
+        };
+        Ok(())
+    }
+
+    fn evaluate(&mut self) -> Result<ScalarValue> {
+        Ok(ScalarValue::Boolean(self.acc))
+    }
+
+    fn size(&self) -> usize {
+        std::mem::size_of_val(self)
+    }
+
+    fn state(&mut self) -> Result<Vec<ScalarValue>> {
+        Ok(vec![ScalarValue::Boolean(self.acc)])
+    }
+
+    fn merge_batch(&mut self, states: &[ArrayRef]) -> Result<()> {
+        self.update_batch(states)
+    }
+
+    fn retract_batch(&mut self, _values: &[ArrayRef]) -> Result<()> {
+        todo!()
+    }
+
+    fn supports_retract_batch(&self) -> bool {
+        todo!()
+    }
+}
+
+// #[cfg(test)]
+// mod tests {
+//     use super::*;
+//     use crate::expressions::col;
+//     use crate::expressions::tests::aggregate;
+//     use crate::generic_test_op;
+//     use arrow::datatypes::*;
+//     use arrow::record_batch::RecordBatch;
+//
+//     #[test]
+//     fn test_bool_and() -> Result<()> {
+//         let a: ArrayRef = Arc::new(BooleanArray::from(vec![true, true, false]));
+//         generic_test_op!(a, DataType::Boolean, BoolAnd, ScalarValue::from(false))
+//     }
+//
+//     #[test]
+//     fn bool_and_with_nulls() -> Result<()> {
+//         let a: ArrayRef = Arc::new(BooleanArray::from(vec![
+//             Some(true),
+//             None,
+//             Some(true),
+//             Some(true),
+//         ]));
+//         generic_test_op!(a, DataType::Boolean, BoolAnd, ScalarValue::from(true))
+//     }
+//
+//     #[test]
+//     fn bool_and_all_nulls() -> Result<()> {
+//         let a: ArrayRef = Arc::new(BooleanArray::from(vec![None, None]));
+//         generic_test_op!(a, DataType::Boolean, BoolAnd, ScalarValue::Boolean(None))
+//     }
+//
+//     #[test]
+//     fn test_bool_or() -> Result<()> {
+//         let a: ArrayRef = Arc::new(BooleanArray::from(vec![true, true, false]));
+//         generic_test_op!(a, DataType::Boolean, BoolOr, ScalarValue::from(true))
+//     }
+//
+//     #[test]
+//     fn bool_or_with_nulls() -> Result<()> {
+//         let a: ArrayRef = Arc::new(BooleanArray::from(vec![
+//             Some(false),
+//             None,
+//             Some(false),
+//             Some(false),
+//         ]));
+//         generic_test_op!(a, DataType::Boolean, BoolOr, ScalarValue::from(false))
+//     }
+//
+//     #[test]
+//     fn bool_or_all_nulls() -> Result<()> {
+//         let a: ArrayRef = Arc::new(BooleanArray::from(vec![None, None]));
+//         generic_test_op!(a, DataType::Boolean, BoolOr, ScalarValue::Boolean(None))
+//     }
+// }
+//
```

Comment:
This test code can be safely removed?

This seems to already [covered by existing sqllogictests here](https://github.com/apache/datafusion/blob/c6b2efccf6238cc87f2414efb28ae3b263ed27af/datafusion/sqllogictest/test_files/aggregate.slt#L3547-L3586).

Reply:
I agree these tests are redundant and can be removed (rather than commented out)

Reply:
Thanks for the confirmation. I'll remove the commented out tests.